packages/sdk-v2/src/sdk/newsfeed/OrbisCachedFeed.ts

```diff
@@ -0,0 +1,116 @@
+import { Orbis } from "@orbisclub/orbis-sdk";
+import { isArray } from "lodash";
+import { createNewsFeedDb } from "./utils";
+import { batch, isValidCID } from "./utils";
+import { AsyncStorage } from "../storage";
+import { IPFSUrls, IpfsStorage } from "../ipfs/sdk";
+
+export type FeedFilter = { context?: string; tag?: string };
+
+export interface FeedPost {
+  id: string;
+  link: string;
+  title: string;
+  content: string;
+  picture: string;
+  published: string;
+  updated: string;
+  sponsored_link: string;
+  sponsored_logo: string;
+}
+
+export class OrbisCachedFeed {
+  IPFS_BATCH = 10;
+  SYNC_PERIOD = 1000 * 60 * 60; //1 hour
+  sdk: Orbis = null;
+  filter: FeedFilter = {};
+  db: any;
+  IPFS: any;
+
+  constructor(filter: FeedFilter, ipfsUrls?: IPFSUrls) {
+    this.db = createNewsFeedDb();
+    this.db.open();
+    this.filter = filter;
+    this.sdk = new Orbis();
+    this.IPFS = new IpfsStorage(ipfsUrls);
```

Comment:
should accept instance of ipfs storage, not instantiate it

Reply:
its fine

Reply:
there is huge issue with architecture
ipfs client is a 'service' class having the same config during whole app lifecycle
such classes should instantiate as singletons (e.g. only one instance per app).
in the feed flow ipfs is just a 3rd party util responsible for loading images (single responsibility rule, interface segregation rule)
it should instantiate separately and have own config. this was a tech definition of the task I've put for @L03TJ3 

please change this

Reply:
ok